ipn/ipnlocal/serve.go

```diff
@@ -447,9 +447,15 @@ func (b *LocalBackend) proxyHandlerForBackend(backend string) (*httputil.Reverse
 		Rewrite: func(r *httputil.ProxyRequest) {
 			r.SetURL(u)
 			r.Out.Host = r.In.Host
+			r.Out.Header.Set("X-Forwarded-Host", r.In.Host)
 			if c, ok := r.Out.Context().Value(serveHTTPContextKey{}).(*serveHTTPContext); ok {
 				r.Out.Header.Set("X-Forwarded-For", c.SrcAddr.Addr().String())
 			}
+			if r.In.TLS == nil {
+				r.Out.Header.Set("X-Forwarded-Proto", "http")
+			} else {
+				r.Out.Header.Set("X-Forwarded-Proto", "https")
```

Comment:
I believe the `X-Forwarded-Proto` will always be `https`. We currently only support an HTTPS listener with `serve`/`funnel`.

Reply:
Thanks for the hint! I added another commit that simplifies the code change by always adding https as the value (= dropping the if).